server/src/controllers/notes.controller.ts

```diff
@@ -0,0 +1,133 @@
+import express, {
+  RequestHandler,
+  Request,
+  Response,
+  NextFunction,
+} from "express";
+import { validationResult } from "express-validator";
+import { validationErrorResponse } from "./utils";
+import mongoose from "mongoose";
+import NoteDocument from "../models/Notes/NotesDocument";
+import NoteCollection from "../models/Notes/NotesCollection";
+
+export const getNotes: RequestHandler = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  // res.status(200).json({ message: "YO" });
+  NoteCollection.find()
+    .exec()
+    .then((notes: NoteDocument[] | null) => {
+      if (!notes) {
+        return Promise.reject(res.status(401).end());
+      }
+
+      res.status(200).json(notes);
+    })
+    .catch((error: Response) => {
+      return next(error);
```

Comment:
Since this controller is the last in the chain, `next()` would be null. We should return a response status of 500 with field error.
```suggestion
      return res.status(500).json({error: error.message});
```